### PR TITLE
Added property_pressure, cleaned up code and added simple TLS support…

### PR DESCRIPTION
**Pull request by mattange**
… for system based certificate authorities (e.g. letsencrypt certificates)

My setup has MQTT behind a reverse proxy with TLS support. Having usernames and passwords in the clear with MQTT is not advisable (if this for instance is used across the internet) so simple TLS support can be added at very low cost.
This assumes the system running the code has certificate authority that is recognised, eg. letsencrypt free certificates (or the MQTT server's CA has distributed public certificates).

In addition while i was investigating the issue with connection, i noticed some lines of code in the paho-mqtt-client code were duplicated. Similarily, the location of the MQTT_SETTINGS structure between 2 files, so i have taken the liberty of removing them.

Hopefully this does not break with other changes you're testing.

### CODE REVIEW

File: homie/mqtt/mqtt_base.py [hunk 1]
```diff
@@ -109,7 +99,7 @@ def _on_message(self, topic, payload, retain, qos):
 
     def _on_disconnect(self, rc):
         logger.warning("MQTT On Disconnect: Result Code {}".format(rc))
-        self.mqtt_connected(False)
+        self.connected = False
```
Comment (mjcumming) [inline homie/mqtt/mqtt_base.py]: there is no property connected - not sure what this change is?

File: homie/mqtt/mqtt_base.py [hunk 2]
```diff
@@ -17,16 +17,6 @@
 logger = logging.getLogger(__name__)
 
 
-MQTT_SETTINGS = {
```
Comment (mjcumming) [inline homie/mqtt/mqtt_base.py:20]: prefer this list of default settings be left in place

File: homie/support/repeating_timer.py [hunk 3]
```diff
@@ -8,34 +8,25 @@
 
 
 class Repeating_Timer(object):
-    DEFAULT_INTERVAL_REDUCTION = 0.9
```
Comment (mjcumming) [inline homie/support/repeating_timer.py:11]: please explain the rationale for removing this

File: homie/support/repeating_timer.py [hunk 4]
```diff
@@ -8,34 +8,25 @@
...
 
     def __init__(self, interval):
-        self.interval = int(interval * self.DEFAULT_INTERVAL_REDUCTION)
-
-        self.start = time.time()
+        self.interval = float(interval)
         self.event = Event()
-
         self.thread = Thread(target=self._target)
-        self.thread.setDaemon(True)
```
Comment (mjcumming) [inline homie/support/repeating_timer.py:22]: why this change?

**Review by mjcumming — COMMENTED** [review]
please review my comments

File: homie/mqtt/mqtt_base.py (hunk 2 above)
Comment (mattange) [reply homie/mqtt/mqtt_base.py:20]: hi - that's fine but i believe the exact same list is available in "homie_mqtt_client.py" which is the one that - i believe - gives the actual defaults to the Paho MQTT derived class. so I'd say it's more appropriate in the other file. else there appears to be a duplication with no particular rationale. Maybe i am mistaken though... although i am also using a few defaults.

Comment (mjcumming) [reply homie/mqtt/mqtt_base.py:20]: found them there, agree, lets delete them here

**Review by mjcumming — APPROVED** [review]

File: homie/support/repeating_timer.py (hunk 4 above)
Comment (mattange) [reply homie/support/repeating_timer.py:22]: hi, sorry this was me being possibly pedantic while going through this (to understand it myself). i was reviewing the documentation at https://docs.python.org/3/library/threading.html#thread-objects and it suggested in this case to use self.thread.daemon = True instead of setDaemon as a function. feel free to reject it would have no impact right now anyway

File: homie/support/repeating_timer.py (hunk 3 above)
Comment (mattange) [reply homie/support/repeating_timer.py:11]: Hi, in this case i really did not understand why it's there in the first place. It appeared to be used in the init function as a multiplier of the actual interval provided in the constructor. so it appeared to me that the interval in seconds would have been actually 0.9 x the interval decided by the user with the prior code. If a float was needed (likely) then i would use python conversion of the integer to float. I have also removed the reference to the _time as that would not be needed anymore.

File: homie/mqtt/mqtt_base.py (hunk 2 above)
Comment (half-monkey) [reply homie/mqtt/mqtt_base.py:20]: Can I suggest adding an (optional) setting to specify where the CA certificate file is? My experience with the PAHO client on Raspberry Pi is that it doesn't find the system-wide default (/etc/ssl/certs/ca-certificates.crt seems to work for me), and would be needed for anyone using self-signed certs for development/personal projects.
